### PR TITLE
[dnm] bump versions for bounty

### PR DESCRIPTION
**Pull request by Reecepbcups**
Collect $500 bounty from Gadakian

gets juno to latest w/ ictest v7.0.0 branch. The commit referenced is not in a release, thus would be cherry picking data. The latest interchaintest v7 release is being used, because that is meant for public consumption.

**VERSIONS ON THE JUNO SIDE**
- cometbft v0.37.4
- Cometbft-db v0.9.0 (unsure if v0.10+ are for v47 or not)
- cosmos-sdk v0.47.10
- IBC v7 v7.3.2
- math v1.3.0

### CODE REVIEW

**Comment by Reecepbcups** [comment]
not linting as that is out of scope

**Comment by Reecepbcups** [comment]
cc @faddat - not linting as that is out of scope. v7 ictest running with Juno on latest as requested.

noble1reece3m8g4m3d0qrpj93rnnseudnpzhr380amd

**Comment by Reecepbcups** [comment]
all ictest passing without issue
image

File: interchaintest/go.mod [hunk 1]
```diff
@@ -10,6 +10,7 @@ replace (
 	// For this nested module, you always want to replace the parent reference with the current worktree.
 	github.com/CosmosContracts/juno/v19 v19.0.0-00010101000000-000000000000 => ../
 	github.com/btcsuite/btcd => github.com/btcsuite/btcd v0.22.2 //indirect
+	github.com/cometbft/cometbft => github.com/cometbft/cometbft v0.37.2
```
Comment (faddat) [inline interchaintest/go.mod:13]: Howdy there, you're replacing comet with old-comet.

The purpose of the exercise was to use the latest dependencies, clearing issues that are solved in the latest comet like the super annoying btcd security warning.  

Thank you for conclusively showing that interchaintest is not kept up to date. 

cometbft's latest release is v0.37.4

File: interchaintest/go.sum [hunk 2]
```diff
@@ -341,10 +341,10 @@ github.com/cockroachdb/redact v1.1.5/go.mod h1:BVNblN9mBWFyMyqK1k3AAiSxhvhfK2oOZ
 github.com/codahale/hdrhistogram v0.0.0-20161010025455-3a0bb77429bd/go.mod h1:sE/e/2PUdi/liOCUjSTXgM1o87ZssimdTWN964YiIeI=
 github.com/coinbase/rosetta-sdk-go/types v1.0.0 h1:jpVIwLcPoOeCR6o1tU+Xv7r5bMONNbHU7MuEHboiFuA=
 github.com/coinbase/rosetta-sdk-go/types v1.0.0/go.mod h1:eq7W2TMRH22GTW0N0beDnN931DW0/WOI1R2sdHNHG4c=
-github.com/cometbft/cometbft v0.37.4 h1:xyvvEqlyfK8MgNIIKVJaMsuIp03wxOcFmVkT26+Ikpg=
```
Comment (faddat) [inline interchaintest/go.sum]: This is what you are removing.  This is the correct version of cometbft, the latest release.

File: interchaintest/go.sum [hunk 3]
```diff
@@ -341,10 +341,10 @@ github.com/cockroachdb/redact v1.1.5/go.mod h1:BVNblN9mBWFyMyqK1k3AAiSxhvhfK2oOZ
...
-github.com/cometbft/cometbft v0.37.4/go.mod h1:Cmg5Hp4sNpapm7j+x0xRyt2g0juQfmB752ous+pA0G8=
-github.com/cometbft/cometbft-db v0.8.0 h1:vUMDaH3ApkX8m0KZvOFFy9b5DZHBAjsnEuo9AKVZpjo=
-github.com/cometbft/cometbft-db v0.8.0/go.mod h1:6ASCP4pfhmrCBpfk01/9E1SI29nD3HfVHrY4PG8x5c0=
+github.com/cometbft/cometbft v0.37.2 h1:XB0yyHGT0lwmJlFmM4+rsRnczPlHoAKFX6K8Zgc2/Jc=
```
Comment (faddat) [inline interchaintest/go.sum:344]: This is what you are adding, a version of cometbft that isn't the latest.

**Comment by faddat** [comment]
Reece, thanks for this pull request that clearly shows that interchaintest v7 doesn't work with standard cosmos ecosystem libraries latest releases.

Since you actually used a replace statement for comet, locking it to a lower-than-latest release, this PR does not win the bounty, but instead illustrates that interchaintest is not compatible with current versions of common cosmos dependencies.

**Comment by Reecepbcups** [comment]
The bounty was to upgrade Juno to use Comet BFT v0.37.4 w/ interchaintest correct?

build then junod version --long :) . it uses cometbft v0.38.4 as the bounty requested using ictest v7.0.0

the pinning does not affect Juno. Thus fully fulling the bounyies request that Juno should be on latest, it is.

**Comment by Reecepbcups** [comment]
good on changing your actual terms of what your actual bounty was. Juno be on latest with interchaintest: working.

Definitly a man of your word: worthless. love you, hugs and kisses. goodnight.

**Comment by faddat** [comment]
Sorry Reece, in your solution, the Juno repository still contains out of date dependencies resulting from interchaintest v7's lack of updates. 

The bounty was not for the creative use of replace statements.

There's no reason that anything in this repository should be held back on account of a test suite.

**Comment by Reecepbcups** [comment]
This PR satisfies the bounty. Will invoice Notional if payment is not paid within 2 weeks.

image

**Comment by Reecepbcups** [comment]
Juno uses the latest. interchaintest may not, but Juno does indeed use the latest and works with interchaintest.

a bounty with only one specific solution is not a bounty.

be a man of your word or kindly, and loveingly with up most care and compassion; fuck off

**Comment by faddat** [comment]
Sir, you added a replace statement to the interchaintest folder in Juno restricting it to a prior version because interchaintest isn't maintained. 

GitHub.com/cosmoscontracts/juno/v20/interchaintest

**Is a part of Juno**

....which is indeed the problem. 

Restricting a key component of Juno to obsolete versions does not satisfy the bounty.

**Comment by Reecepbcups** [comment]
Is it what is running on the Juno binary? NO. CometBFT V0.34.4 is used on the Juno binary

fuck off with your technical standards that have no actual impact on reality. 

Any real dev would know this. emphasis on real dev. you know, people that actually can code in cosmos.

**Comment by Reecepbcups** [comment]
crazy I now have negative respect for a person

congrats on being a first

**Comment by faddat** [comment]
Sir, this repository is called Juno.

Juno's code is in here.  

I'm talking about Juno, you're talking about me. 

Are you able to do this so that there are no references to obsolete code in this repository while using that commit?

That would satisfy the bounty.

**Pull request closed without merging**